lib/src/file_util.rs

```diff
@@ -256,4 +281,14 @@ mod tests {
 
         assert!(persist_content_addressed_temp_file(temp_file, &target).is_ok());
     }
+
+    #[test]
+    fn test_copy_async_to_sync() {
+        let input = b"hello";
+        let mut output = vec![];
+
+        let result = copy_async_to_sync(Cursor::new(&input), &mut output).block_on();
+        assert!(result.is_ok());
+        assert_eq!(result.unwrap(), 5);
```

Comment:
nit: can you add test for data bigger than the buffer size?

Reply:
Makes sense. Done.